### PR TITLE
Prefix cluster name during testing with username

### PR DESCRIPTION
**Pull request by deepakm-ntnx**
**What this PR does / why we need it**:
now creating user specific name of cluster
    
    this will avoid tests running by multiple users on same setup from
    conflicting

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**How Has This Been Tested?**:

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and test output_

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Review by tuxtof — APPROVED** [review]

**Review by yannickstruyf3 — CHANGES_REQUESTED** [review]
Let's generate a random string instead of using username

**Review by swetangfinviya — COMMENTED** [review]
For me make print-capx-controller-logs is not working. The variable CAPX_CONTROLLER_POD_NAME is not getting created.

Here is the output:
swetang.finviya@C02X740FJHD2 cluster-api-provider-nutanix % make print-capx-controller-logs
CAPX_CONTROLLER_POD_NAME= && kubectl -n capx-system logs -f  manager
Error from server (NotFound): pods "manager" not found
make: *** [print-capx-controller-logs] Error 1

**Review by swetangfinviya — COMMENTED** [review]
The following make command works for me:
`.PHONY: print-capx-controller-logs
print-capx-controller-logs: ## logs the controller pod output with -f mode
	kubectl -n capx-system logs -f $(shell kubectl -n capx-system get pods | grep capx-controller | awk '{print $$1}') manager`

**Review by tuxtof — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -140,7 +140,7 @@ GINKGO_NOCOLOR ?= false
 FLAVOR ?= e2e
 
 TEST_NAMESPACE=capx-test-ns
-TEST_CLUSTER_NAME=mycluster
+TEST_CLUSTER_NAME=capx-cl-${USER}
```
Comment (yannickstruyf3) [inline Makefile:143]: I would suggest to remove the user parameter and just allow the user to override TEST_CLUSTER_NAME and TEST_NAMESPACE variables

**Review by yannickstruyf3 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (deepakm-ntnx) [reply Makefile:143]: could you please elaborate on your concern with usernames? 
i can use value with username as a default value but give preference to env variable if set. does that sound ok?

**Comment by deepakm-ntnx** [comment]
closing it due to merge conflicts

**Pull request closed without merging**